build.sh

```diff
@@ -237,6 +237,9 @@ elif [ "$build_all" != true ] && [ -z "$build_managed$build_nodejs$build_java$bu
     # We believe the most common thing our contributors will work on is C#, so if no other build group was picked, build the C# projects.
     __warn "No default group of projects was specified, so building the 'managed' and its dependent subset of projects. Run ``build.sh --help`` for more details."
     build_managed=true
+elif [ "$build_all" != true ] && [ -z "$build_managed" ] && ! [[ "$build_nodejs$build_java$build_native$build_installers" =~ "true" ]]; then
```

Comment:
FYI the check above is a bit different than what we do in build.ps1. Here, we use one variable for `--build-xyz` and `--no-build-xyz` whereas Powershell keeps them separate 'til much later in the script. Adding this extra block makes things more similar.

Note builds can still fall back to attempting to build all of AspNetCore.sln e.g. `./build.ps1 --noBuildManaged` ☹️ Fortunately, that's a corner case.